contrib/win32/win32compat/w32log.c

```diff
@@ -35,46 +35,45 @@
 
 #include "inc\syslog.h"
 #include "misc_internal.h"
+#include "..\..\..\config.h"
+#include "inc\utf.h"
 
 #define MSGBUFSIZ 1024
 static int logfd = -1;
 
 /*
- * open a log file using the name of executable under logs folder
- * Ex. if called from c:\windows\system32\openssh\sshd.exe
- * logfile - c:\windows\system32\openssh\logs\sshd.log
+ * log file location will be - "%programData%\\openssh\\logs\\<module_name>.log"
  */
 void
 openlog(char *ident, unsigned int option, int facility)
-{
-	wchar_t *logs_dir = L"\\logs\\";
+{	
 	if (logfd != -1 || ident == NULL)
 		return;
 
-	wchar_t path[PATH_MAX] = { 0 }, log_file[PATH_MAX + 12] = { 0 };
-	errno_t r = 0;
-	if (GetModuleFileNameW(NULL, path, PATH_MAX) == 0)
-		return;
+	wchar_t *logs_dir = L"\\logs\\";
+	wchar_t module_path[PATH_MAX] = { 0 }, log_file[PATH_MAX + 12] = { 0 };
 
-	path[PATH_MAX - 1] = L'\0';
+	if (GetModuleFileNameW(NULL, module_path, PATH_MAX) == 0)
+		return;
 
-	if (wcsnlen(path, MAX_PATH) > MAX_PATH - wcslen(logs_dir))
+	if (wcsnlen(module_path, MAX_PATH) > MAX_PATH - wcslen(logs_dir))
 		return;
 
 	/* split path root and module */
 	{
-		wchar_t* tail = path + wcsnlen(path, MAX_PATH);
-		while (tail > path && *tail != L'\\' && *tail != L'/')
+		wchar_t* tail = module_path + wcsnlen(module_path, MAX_PATH);
+		while (tail > module_path && *tail != L'\\' && *tail != L'/')
 			tail--;
+			
+		wchar_t* ssh_root_path_w = utf8_to_utf16(get_ssh_dir_path()); // "%programData%\\openssh"
```

Comment:
update the comments to follow the code guidance

Reply:
fixed